lib/immobilienscout/request_error_handler.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Immobilienscout
+  module RequestErrorHandler
+    class << self
+      ERROR_RESOURCE_NOT_FOUND = 'ERROR_RESOURCE_NOT_FOUND'
+      ERROR_COMMON_RESOURCE_NOT_FOUND = 'ERROR_COMMON_RESOURCE_NOT_FOUND'
+      ERROR_RESOURCE_VALIDATION = 'ERROR_RESOURCE_VALIDATION'
+
+      def handle(parsed_response)
+        if parsed_response.messages.is_a?(Array)
+          error_code = parsed_response.messages&.first&.code
+          error_message = parsed_response.messages.map(&:messages)
```

Comment:
this is probably `error_messages` since it's an array ;)

Reply:
Yup. Fixed it. Thanks for spotting it.